test cases/common/132 get define/meson.build

```diff
@@ -2,49 +2,50 @@ project('get define', 'c', 'cpp')
+  'sunos'     : ['__sun__',       '1'],
+
+  # The __FreeBSD__ define will be equal to the major version of the release
+  # (ex, in FreeBSD 11.x, __FreeBSD__ == 11). To make the test robust when
+  # being run on various versions of FreeBSD, just test that the define is
+  # set.
+  'freebsd'   : ['__FreeBSD__'],
+}
+
 foreach lang : ['c', 'cpp']
   cc = meson.get_compiler(lang)
-  if host_system == 'linux'
-    d = cc.get_define('__linux__')
-    assert(d == '1', '__linux__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'darwin'
-    d = cc.get_define('__APPLE__')
-    assert(d == '1', '__APPLE__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'windows'
-    d = cc.get_define('_WIN32')
-    assert(d == '1', '_WIN32 value is @0@ instead of 1'.format(d))
-  elif host_system == 'cygwin'
-    d = cc.get_define('__CYGWIN__')
-    assert(d == '1', '__CYGWIN__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'haiku'
-    d = cc.get_define('__HAIKU__')
-    assert(d == '1', '__HAIKU__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'freebsd'
-    # the __FreeBSD__ define will be equal to the major version of the release
-    # (ex, in FreeBSD 11.x, __FreeBSD__ == 11). To make the test robust when
-    # being run on various versions of FreeBSD, just test that the define is
-    # set.
-    d = cc.get_define('__FreeBSD__')
-    assert(d != '', '__FreeBSD__ value is unset')
-  elif host_system == 'dragonfly'
-    d = cc.get_define('__DragonFly__')
-    assert(d == '1', '__DragonFly__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'netbsd'
-    d = cc.get_define('__NetBSD__')
-    assert(d == '1', '__NetBSD__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'openbsd'
-    d = cc.get_define('__OpenBSD__')
-    assert(d == '1', '__OpenBSD__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'gnu'
-    d = cc.get_define('__GNU__')
-    assert(d == '1', '__GNU__ value is @0@ instead of 1'.format(d))
-  elif host_system == 'sunos'
-    d = cc.get_define('__sun__')
-    assert(d == '1', '__sun__ value is @0@ instead of 1'.format(d))
-  else
+
+  if not system_define_map.has_key(host_system)
     error('Please report a bug and help us improve support for this platform')
   endif
 
+  system_define = system_define_map.get(host_system)
+
+  def_name  = system_define[0]
+  def_val   = cc.get_define(system_define[0])
+  def_exist = cc.has_define(system_define[0])
+
+  assert((def_val != '') == def_exist,
+   'The has_define and get_define results for @0@ disagree with each other'.format(def_name))
+
+  if system_define.length() == 2
+    assert(def_val == system_define[1],
+      '@0@ value is @1@ instead of @2@'.format(def_name, def_val, system_define[1]))
+  elif system_define.length() == 1
+    assert(def_val != '', '@0@ value is unset'.format(def_name))
+  else
+    assert('Invalid number of items in system_define array, this is a bug in the test!')
```

Comment:
Technically I think this doesn't do anything -- the assert is always true, because args[0] is a string that isn't empty, so nothing prints. The error() function  makes more sense here.

Reply:
Oops indeed, I meant to use that here…

Reply:
> the assert is always true, because args[0] is a string that isn't empty, so nothing prints

At some point we changed `assert` to require bool arguments, so it will actually error out.